lib/util/exec/common.ts

```diff
@@ -37,16 +34,84 @@ export interface DockerOptions {
   postCommands?: DockerExtraCommands;
 }
 
-export interface RawExecOptions extends ChildProcessExecOptions {
-  encoding: string;
-}
+export type RawExecOptions = SpawnOptions;
 
 export interface ExecResult {
   stdout: string;
   stderr: string;
+  output: string;
+}
+
+interface ChildProcess {
+  stdout: Readable;
+  stderr: Readable;
+  on(event: string | symbol, listener: (...args: any[]) => void): this;
+}
+
+function pipeStreamToBuffers(stream: Readable, ...buffers: string[][]): void {
+  stream.setEncoding('utf-8');
+  stream.on('data', (s) => {
+    buffers.forEach((buf) => buf.push(s));
+  });
+}
+
+export class ExecError extends Error {
+  constructor(
+    message: string,
+    public stdout: string,
+    public stderr: string,
+    public output: string,
```

Comment:
We need to make sure `output` is sanitized for logging, I think we can simply remove it, otherwise we have a lot duplicate log output

https://github.com/renovatebot/renovate/blob/e8c80fbe625acf1304ac9da8ca979e82c24e14ec/lib/logger/err-serializer.ts#L41